tests/ha/test_multi_server_pod_failure.py

```diff
@@ -285,3 +289,330 @@ def test_reads_kserver_pods_fail(self):
 
         LOGGER.info("Completed: Test to verify degraded READs after all K server pods "
                     "down - unsafe shutdown.")
+
+    # pylint: disable=too-many-statements
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-40568")
+    @CTFailOn(error_handler)
+    def test_reads_till_kserver_pods_fail(self):
+        """
+        Test to verify degraded READs after each pod failure till K server pods "
+        "down - unsafe shutdown.
+        """
+        LOGGER.info("Started: Test to verify degraded READs after each pod failure till K "
+                    "server pods down - unsafe shutdown.")
+
+        LOGGER.info("STEP 1: Perform WRITEs with variable object sizes. 0B + (1KB - 512MB)")
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.test_prefix = 'test-40568'
+        self.s3_clean = users
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipread=True,
+                                                    skipcleanup=True, nsamples=2, nclients=2)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed WRITEs with variable sizes objects.")
+
+        LOGGER.info("Step 2: Perform READs and verify DI on the written data")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipwrite=True,
+                                                    skipcleanup=True, nsamples=2, nclients=2)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 2: Performed READs and verified DI on the written data")
+
+        LOGGER.info("Shutdown %s (K) server pods one by one and verify read/verify "
+                    "after each pod down", self.kvalue)
+        LOGGER.info("Get server pod names to be deleted")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.SERVER_POD_NAME_PREFIX)
```

Comment:
instead of calling this function everytime, can we add one variable in setup method say data_pods and then copy the same every time to new list. 
in setup:
        self.data_pods = self.node_master_list[0].get_all_pods(pod_prefix=const.POD_NAME_PREFIX)
        self.server_pods = self.node_master_list[0].get_all_pods(
            pod_prefix=const.SERVER_POD_NAME_PREFIX)

in tests as per need of required list do copy,
ie. pod_list = self.data_pods.copy()

@ApurwaMohite  @sonalk0209  Please suggest since we are developing this file whole new, its easy to change.

Reply:
yeah can be done but what will be the benifit?

Reply:
> yeah can be done but what will be the benifit?

Just we don't need to call get_all_pods every time in test if we are going to add check in all **pods status** and **pods service** otherwise we will need to fetch both list (data and server pods) everytime in the test.

Reply:
We have to add data_pods + server_pods to get all pods status.